dlup/data/dataset.py

```diff
@@ -27,9 +27,10 @@
 
 T_co = TypeVar("T_co", covariant=True)
 T = TypeVar("T")
-_BaseAnnotationTypes = SlideImage | WsiAnnotations
-_AnnotationTypes = list[tuple[str, _BaseAnnotationTypes]] | _BaseAnnotationTypes
-_LabelTypes = str | bool | int | float
+_BaseAnnotationTypes = Union[SlideImage, WsiAnnotations]
+_AnnotationTypes = Union[list[tuple[str, _BaseAnnotationTypes]], _BaseAnnotationTypes]
+_LabelTypes = Union[str, bool, int, float]
+ROIType = tuple[tuple[tuple[int, int], tuple[int, int]], ...]
 
```

Comment:
in the original mask-to-polygon PR you refactored many `Union`s into `x | y`s. Why are we doing the reverse here?

Reply:
Because python 3.9 still has issues with these, and many people seem to use that

Reply:
DLUP now is >=3.10 though, no? 

`python_requires=">=3.10",` in `setup.py`

Reply:
@jonasteuwen Should we care about this or leave it as is?